cookbooks/bcpc-hadoop/recipes/graphite_to_zabbix.rb

```diff
@@ -29,156 +31,144 @@
 ruby_block "zabbix_monitor" do
   block do
     require 'zabbixapi'
-    #
+    
     # Make connection to zabbix api url
-    #
     zbx=ZabbixApi.connect(:url => "https://#{node['bcpc']['management']['vip']}:#{node['bcpc']['zabbix']['web_port']}/api_jsonrpc.php", 
                           :user => 'admin', 
                           :password => "#{get_config('zabbix-admin-password')}")
     if zbx.nil?
       Chef::Log.fatal!("Fatal error: could not connect to Zabbix server")
     end
-    #
-    # Looping through the graphite queries attribute to define the required zabbix hosts, items and triggers
-    #
+
+    # Fetch graphite hosts
     graphite_hosts = (get_node_attributes(MGMT_IP_ATTR_SRCH_KEYS,"graphite","bcpc").map {|v| v['mgmt_ip']}).join(",")
+    if graphite_hosts.empty?
+      Chef::Log.fatal!("Fatal error: No graphite hosts found")
```

Comment:
Logical thing would be to *not* continue. Same would be true for the `zbx.nil?` check